hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/access/NonTxInvalidationCacheAccessDelegate.java

```diff
@@ -7,17 +7,36 @@
 package org.infinispan.hibernate.cache.commons.access;
 
 import org.hibernate.cache.CacheException;
+import org.infinispan.commands.CommandsFactory;
+import org.infinispan.commands.write.PutKeyValueCommand;
+import org.infinispan.commands.write.RemoveCommand;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.context.impl.FlagBitSets;
+import org.infinispan.factories.ComponentRegistry;
 import org.infinispan.hibernate.cache.commons.impl.BaseRegion;
 import org.hibernate.cache.spi.access.SoftLock;
+import org.infinispan.interceptors.AsyncInterceptorChain;
+import org.infinispan.metadata.EmbeddedMetadata;
+import org.infinispan.metadata.Metadata;
 
 /**
  * Delegate for non-transactional caches
  *
  * @author Radim Vansa &lt;rvansa@redhat.com&gt;
  */
 public class NonTxInvalidationCacheAccessDelegate extends InvalidationCacheAccessDelegate {
+	private final AsyncInterceptorChain invoker;
+	private final CommandsFactory commandsFactory;
+	private final Metadata metadata;
+
 	public NonTxInvalidationCacheAccessDelegate(BaseRegion region, PutFromLoadValidator validator) {
 		super(region, validator);
+		ComponentRegistry cr = region.getCache().getComponentRegistry();
```

Comment:
this was the kind of code I didn't want to see :(

I'm not sure if there are other users who are using ThreadLocal and pointing them this code if their code stops working doesn't seem right

I guess we don't have a choice right?

Reply:
2LC implementation does worse things than this. I haven't changed any Infinispan API. And it seems to me that this is better than actually tunneling any information between different components through `ThreadLocal`.

There can't be other consumers of the ThreadLocal - note that if anyone started to hack 2LC even more, the PFLV has changed package so it's broken anyway. IMO the 2LC is not meant as anything 'extensible' (beyond e.g. wfly integration), it's really an implementation of component, not a library.